projects/common/src/utilities/validators/domain-validator.ts

```diff
@@ -0,0 +1,7 @@
+export const DOMAIN_REGEX = new RegExp(/^[a-z0-9\-]{3,}\.[a-z]{2,}$/);
```

Comment:
Validating domains is _hard_ - do we have a requirement to do so? There's almost always domains that can get around it. This reads like it requires all lower case, ascii characters, but all kinds of fun stuff can show up in domain names. Country/second level domains (e.g. `co.uk`) I believe would fail this. Non english characters, upper cases, subdomains (e.g. `hyper.hypertrace.org`) would all fail here, too.

In the past, I've generally worked under the philosophy of undervalidating rather then over.

Reply:
I should probably name it differently. The function is intended to validate the `domain` part of an email.
Will make the change to accommodate domains like `.co.uk`

Reply:
all of the cases I described are what you're saying. `aaron@hyper.hypertrace.org` is a valid email (well not that specifically, but that structure). Email addresses aren't case sensitive (at least I don't think they normally are), so `AARON@HYPERTRACE.ORG` is valid too, and likewise, the non-english characters can show up there as far as I'm aware.

Reply:
my suggestion from slack - `^.+\.[a-zA-Z]{2,}$` - basically adds upper case support and gives up on trying to validate anything before the TLD, just makes sure it's at least one char.

Reply:
Updated. Thanks!